### PR TITLE
[TTAHUB-3006] Show validation/error message when attempting to enter a generic iPD URL

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Disallow the following URLs from being entered as resources

- https://eclkc.ohs.acf.hhs.gov/professional-development/individualized-professional-development-ipd-portfolio/course-catalog
- https://eclkc.ohs.acf.hhs.gov/professional-development/individualized-professional-development-ipd-portfolio/individualized-professional-development-ipd-portfolio
- https://eclkc.ohs.acf.hhs.gov/cas/login

## How to test
- Update ttahub/common
- Enter in the invalid URLS in the AR objective form (session and activity report)
- Confirm that they cannot be saved

## Issue(s)
* https://jira.acf.gov/browse/TTAHUB-3006

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
@AdamAdHocTeam 

> When I enter one of the invalid resources I see the below error message. Do we want to show a different message as the URL is correctly formed but invalid?

Fixed, thanks

File: packages/common/src/constants.js [hunk 1]
```diff
@@ -364,3 +364,20 @@ const REOPEN_REASONS = {
...
+  url: 'https://eclkc.ohs.acf.hhs.gov/professional-development/individualized-professional-development-ipd-portfolio/individualized-professional-development-ipd-portfolio',
+  error: 'This link is no longer accepted in this field. Enter iPD courses used during your TTA session in the other field in this section.',
+}, {
+  url: 'https://eclkc.ohs.acf.hhs.gov/cas/login',
+  error: 'This link is no longer accepted in this field. Enter iPD courses used during your TTA session in the other field in this section.',
+}];
+
+exports.DISALLOWED_URLS = DISALLOWED_URLS;
+
+const VALID_URL_REGEX = /(?<url>(?<scheme>http(?:s)?):\/\/(?:(?<user>[a-zA-Z0-9._]+)(?:[:](?<password>[a-zA-Z0-9%._\+~#=]+))?[@])?(?:(?:www\.)?(?<host>[-a-zA-Z0-9%._\+~#=]{1,}\.[a-z]{2,6})|(?<ip>(?:[0-9]{1,3}\.){3}[0-9]{1,3}))(?:[:](?<port>[0-9]+))?(?:[\/](?<path>[-a-zA-Z0-9'@:%_\+.,~#&\/=()]*[-a-zA-Z0-9@:%_\+.~#&\/=()])?)?(?:[?](?<query>[-a-zA-Z0-9@:%_\+.~#&\/=()]*))*)/ig;
```
Comment (AdamAdHocTeam) [inline packages/common/src/constants.js:381]: what could go wrong lol

**Review by AdamAdHocTeam — COMMENTED** [review]
Which link is not longer accepted?

image

**Review by AdamAdHocTeam — APPROVED** [review]
Overall looks good. Not sure if you want to iron out the language for the invalid URLS.

Maybe just a UL below the message saying the offending URLS.

**Pull request merged**